### PR TITLE
Feat/price feed factory

### PR DESCRIPTION
**Pull request by jamesondh**

### CODE REVIEW

File: contracts/FluxPriceFeedFactory.sol [hunk 1]
```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts/access/AccessControl.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+import "./interface/IERC2362.sol";
+import "./FluxPriceFeed.sol";
+import "hardhat/console.sol";
```
Comment (mariocao) [inline contracts/FluxPriceFeedFactory.sol:8]: I think we should use this only for debugging. :)

Some `console.log` should be removed.

File: contracts/FluxPriceFeedFactory.sol [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+ * @author fluxprotocol.org
+ */
+contract FluxPriceFeedFactory is AccessControl, IERC2362 {
+    bytes32 public constant VALIDATOR_ROLE = keccak256("VALIDATOR_ROLE");
+
+    // mapping of id to FluxPriceFeed
+    mapping(bytes32 => FluxPriceFeed) public fluxPriceFeeds;
+
+    event FluxPriceFeedCreated(bytes32 indexed id, address indexed oracle);
+    event Log(string message);
```
Comment (mariocao) [inline contracts/FluxPriceFeedFactory.sol:21]: In `FluxPriceFeed.sol` we added some code doc to events. Should we do the same?

File: contracts/FluxPriceFeedFactory.sol [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+                    // transmission is successful, nothing to do
+                } catch Error(string memory reason) {
+                    // catch failing revert() and require()
+                    console.log("ERRORR");
+                    emit Log(reason);
+                }
+                // else create an oracle then transmit values
+            } else {
+                _deployOracle(id, _pricePairs[i], _decimals[i]);
+                fluxPriceFeeds[id].transmit(_answers[i]);
```
Comment (mariocao) [inline contracts/FluxPriceFeedFactory.sol:82]: I wonder if we should change the logic of this `if-else` statements to unify the `transmit` call.

```solidity
if (address(fluxPriceFeeds[id]) == address(0x0) {
    _deployOracle(id, _pricePairs[i], _decimals[i]);
}

try fluxPriceFeeds[id].transmit(_answers[i]) {
    // transmission is successful, nothing to do
} catch Error(string memory reason) {
    // catch failing revert() and require()
    emit Log(reason);
}
```

File: contracts/FluxPriceFeedFactory.sol [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+    /**
+     * @notice returns address of a price feed id
+     * @param _id hash of the price pair string to query
+     */
+    function addressOfPricePair(bytes32 _id) external view returns (address) {
+        return address(fluxPriceFeeds[_id]);
+    }
+
+    function typeAndVersion() external pure virtual returns (string memory) {
```
Comment (mariocao) [inline contracts/FluxPriceFeedFactory.sol:131]: Missing code doc

File: test/factory/FluxPriceFeedFactory.behavior.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  // });
+
+  // it("should revert if transmitted arrays aren't equal", async function () {
+  //   let pricePairs = [this.eth_usd_str, this.btc_usd_str];
+  //   let decimals = [3, 3];
+  //   let answers = [3000];
+  //   await expect(this.factory.connect(this.signers.admin).transmit(pricePairs, decimals, answers)).to.be.revertedWith(
+  //     "Transmitted arrays must be equal",
+  //   );
+  // });
```
Comment (mariocao) [inline test/factory/FluxPriceFeedFactory.behavior.ts:66]: Why are they commented?

**Review by mariocao — COMMENTED** [review]
Great!!!! :fireworks: 

Please double check the errors shown after calling `yarn lint`.

File: contracts/FluxPriceFeedFactory.sol (hunk 3 above)
Comment (mennatabuelnaga) [reply contracts/FluxPriceFeedFactory.sol]: Just did that!

**Review by mariocao — APPROVED** [review]

**Pull request merged**